CronRepeatOptions.repeat is wrongly enforcing declaration of the 'key' property in TS

<!--
You may report several types of issues. Bug reports, enhancements or questions.
For bug reports however you are required to provide some information so that the 
issue can be resolved efficiently. The following template should be filled for bugs.

Before submitting the bug just think twice if you really need to submit the bug
or you may have some issue in your own code, remember that handling issues is
time consuming, would you better like that we spend time improving the library
or on non-issues :).
-->

## Description
The 4.11 release, specifically the ([e4e6457](https://github.com/OptimalBits/bull/commit/e4e64572a3ad259d9cb90d5dec81e8565eeadca1)) commit, introduced a requirement in the `repeat` property of the `CronRepeatOptions` TS type to set a `readonly key` property. This is requiring us to declare the key property to add scheduled jobs. For example, if we try to add a job with the opts `{ repeat: { cron: "* * * * *" } }` the TS compiler will fail with the following message:

```
error TS2322: Type '{ cron: string; }' is not assignable to type '((CronRepeatOptions | EveryRepeatOptions) & { readonly key: string; }) | undefined'.
  Type '{ cron: string; }' is not assignable to type 'CronRepeatOptions & { readonly key: string; }'.
     Property 'key' is missing in type '{ cron: string; }' but required in type '{ readonly key: string; }'.
```

According to the documentation this property is not set by the user, but by Bull. So this type declaration should be modified to conform with the documentation.


## Minimal, Working Test code to reproduce the issue.
Try sending a scheduled job without the key property in TS and the compilation error will appear.

## Bull version
4.11

## Additional information
Introduced by this commit:
> **types**: add missing keys to repeat opts ([e4e6457](https://github.com/OptimalBits/bull/commit/e4e64572a3ad259d9cb90d5dec81e8565eeadca1))

--- a/index.d.ts
+++ b/index.d.ts
@@ -447,7 +447,7 @@ declare namespace Bull {
           /**
            * The key for the repeatable job metadata in Redis.
            */
-          readonly key: string;
+          readonly key?: string;
         })
       | undefined;
 

